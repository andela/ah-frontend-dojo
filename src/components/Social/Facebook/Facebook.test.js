// react libraries
import React from 'react';
// third-party libraries
import { shallow } from 'enzyme';
// components
import FacebookButton from './index';
// Define a shared shallow renderer function
const fbButton = (props = {}) => {
  const facebook = shallow(<FacebookButton {...props} />);
  return facebook;
};

describe('Facebook Login', () => {
// Pass the value of the shallow renderer before each test
  let facebook;
  beforeEach(() => {
    facebook = fbButton();
  });

  it('should render without errors', () => {
    const wrapper = shallow(<FacebookButton />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should work as expected', () => {
    const response = {
      accessToken: 'testToken',
    };
    facebook.instance().responseFacebook(response);
  });

  it('should render a facebook button', () => {
    const wrapper = facebook.find('.facebookbutton');
    expect(wrapper.length).toBe(1);
  });
});
