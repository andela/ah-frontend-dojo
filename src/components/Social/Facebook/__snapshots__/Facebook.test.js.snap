// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facebook Login should render without errors 1`] = `
<div
  className="facebookbutton"
>
  <t
    appId="279362183016248"
    callback={[Function]}
    cssClass="kep-login-facebook"
    fields="name,email,picture"
    icon={
      <img
        alt="Facebook"
        height="30px"
        src="https://img.icons8.com/color/48/000000/facebook-circled.png"
        widith="30px"
      />
    }
    size="metro"
    tag="button"
    textButton={false}
    typeButton="button"
  />
</div>
`;
